### PR TITLE
(IAC-383) Fixed Logging and monitoring in a K8s 1.22 deployment

### PR DESCRIPTION
**Pull request by riragh**
### Fixed Logging and monitoring in a K8s 1.22 deployment.:
- Updated the annotations required for logging and monitoring to work with K8s v1.22
### Test coverage:
- Verified the changes on Azure with
    - K8s v1.22.4 and INGRESS_NGINX_CHART_VERSION: 4.0.13
    - K8s v1.19.13 and INGRESS_NGINX_CHART_VERSION: 3.20.1
- Verified the changes on GCP with
    - K8s 1.22.3-gke.700 with INGRESS_NGINX_CHART_VERSION: 4.0.13

### CODE REVIEW

File: roles/monitoring/templates/user-values-elasticsearch-open.yaml [hunk 1]
```diff
@@ -4,8 +4,11 @@ kibana:
     nodePort: null
   ingress:
     annotations:
+      kubernetes.io/ingress.class: nginx
       nginx.ingress.kubernetes.io/backend-protocol: HTTPS
     enabled: true
+    # ingressClassName is not currently supported by the Open Distro helm chart
+    # ingressClassName: nginx
```
Comment (thpang) [inline roles/monitoring/templates/user-values-elasticsearch-open.yaml:11]: Are these needed for the future or place holders?

File: roles/monitoring/templates/user-values-elasticsearch-open.yaml [hunk 2]
```diff
@@ -25,7 +28,10 @@ elasticsearch:
     ingress:
       annotations:
         nginx.ingress.kubernetes.io/backend-protocol: HTTPS
+        kubernetes.io/ingress.class: nginx
       enabled: true
+      # ingressClassName is not currently supported by the Open Distro helm chart
+      # ingressClassName: nginx
```
Comment (thpang) [inline roles/monitoring/templates/user-values-elasticsearch-open.yaml:34]: Are these needed for the future or place holders?

**Review by thpang — CHANGES_REQUESTED** [review]
Just a couple of comments for clarification.

File: roles/monitoring/templates/user-values-elasticsearch-open.yaml (hunk 1 above)
Comment (riragh) [reply roles/monitoring/templates/user-values-elasticsearch-open.yaml:11]: Yes, these were place holders in the sample files provided by the viya4-monitoring-kubernetes, I kept them same in our files.

File: roles/monitoring/templates/user-values-elasticsearch-open.yaml (hunk 2 above)
Comment (riragh) [reply roles/monitoring/templates/user-values-elasticsearch-open.yaml:34]: Yes, these were place holders in the sample files provided by the viya4-monitoring-kubernetes, I kept them same in our files.

**Review by thpang — APPROVED** [review]
LGTM

**Comment by thpang** [comment]
Do these changes work with both ingress-ngnix v0.4.3 and v1.1.0? Checking because if these are merged into main it will break folks using the older version. If they are breaking changes. We'll have to set aside some time to do the release and sync things.

**Comment by riragh** [comment]
> 

Yes this changes work for both k8s v1.19 (the default) with INGRESS_NGINX_CHART_VERSION: 3.20.1
and k8s v1.22 with INGRESS_NGINX_CHART_VERSION: 4.0.13

**Review by sayeun — APPROVED** [review]
Reviewed the 2 files with Ritika.  LGTM too.  👍

**Review by jarpat — APPROVED** [review]
Tested this in GKE, looks good.
K8s 1.22.3-gke.700 with INGRESS_NGINX_CHART_VERSION: 4.0.13.

**Comment by thpang** [comment]
Did you do a squash and merge or just a merge? When you do the final merge into `main` the process was to `squash and merge` items to keep the push history clean. If needed you can alter based on your processes.

**Comment by thpang** [comment]
Also be sure to delete the working branches when the final merge takes place.

**Comment by riragh** [comment]
> Did you do a squash and merge or just a merge? When you do the final merge into `main` the process was to `squash and merge` items to keep the push history clean. If needed you can alter based on your processes.

I think I just used merge. Will squash and merge here after.

**Pull request merged**